### DISCUSSION TITLE
Bank Accounts are repeating themselves under Bankreconciliation.php

### DISCUSSION THREAD

**Discussion by wissedenis** [timschofield/webERP#324] [Problems / Bugs]
<img width="212" alt="image" src="https://github.com/user-attachments/assets/6bf9cdf0-2dac-42c1-b7c0-0e859ef70870" />
Hello @timschofield 
I am inquiring why does the banks get repeated under under Bank Reconciliation
from line 125 to 140
Duplicate option Output

```
else {
	while ($MyRow=DB_fetch_array($AccountsResults)) {

		// Lists bank accounts order by name
		echo '<option',
			((isset($_POST['BankAccount']) and $_POST['BankAccount'] == $MyRow['accountcode']) ? ' selected="selected"' : '' ),
			' value="', $MyRow['accountcode'], '">', $MyRow['bankaccountname'], ' - ', $MyRow['currcode'], '</option>';


		/*list the bank account names */
		if (isset($_POST['BankAccount']) and $_POST['BankAccount']==$MyRow['accountcode']) {
			echo '<option selected="selected" value="' . $MyRow['accountcode'] . '">' . $MyRow['bankaccountname'] . '</option>';
		} else {
			echo '<option value="' . $MyRow['accountcode'] . '">' . $MyRow['bankaccountname'] . '</option>';
		}
	}
	echo '</select></td>
		</tr>';
}
```

there are two  echo options called and merged under the `'</select>` hence getting lists repeated 
one option is appended with: `' - ', $MyRow['currcode'], ` and the other is not.
from line 130 it goes straight to line 137, this makes the banks repeated

**Comment by pakricard** [comment]
Hi @wissedenis : Thanks for reporting. Moved to the issues section, as there is not much to discuss, it's obviously a bug. ;-)

**Reply by wissedenis** [reply]
> Hi @wissedenis : Thanks for reporting. Moved to the issues section, as there is not much to discuss, it's obviously a bug. ;-)

where do I push or share my modified files.
Php 8.1+

**Reply by pakricard** [reply]
Hi:
I committed this, on your behalf. 
For future collaborations, please ask @timschofield (as he owns the master repository). You can then send a pull request, and he will commit into the main branch.
Many thanks

**Reply by dalers** [reply]
> where do I push or share my modified files.

Thank you very mucvh @wissedenis for your contribution (and thank you @pakricard for committing).

If you would like to see your name in the project Git log for future submissions, you will need to first fork this repo in GitHub, commit changes to your fork and send a Pull Request ("PR") from your fork to the master repo (this repo). There is no need to seperately ask @timschofield when you send a pull request, that's the purpose of the pull request. There is a good overview on the webERP website https://weberp.org/Development.html.

In a perfect world, an Issue is generally created first with a description of the incorrect behavior and proposed solution and the description in the pull request references the issue (since the pull request is to resolve the issue). Create a Discussion topic when discussion is required to confirm the behavior is in fact an issue, and to decide on the best resolution.
